### ISSUE TITLE
Buffer polyfill not work on ios lower then 14

### ISSUE THREAD

**Issue by fraitag** [vercel/next.js#66115]
### Link to the code that reproduces this issue

https://codesandbox.io/p/devbox/floral-snowflake-cyrx5f

### To Reproduce

Add something with Buffer on client side
Try open page on IOS lower then 14
Console report the error:
![image](https://github.com/vercel/next.js/assets/17552716/6c7b4b77-3c57-4c4f-a397-5d1a0f3696cb)

### Current vs. Expected behavior

The polyfill should work on versions supported by Next.js.

### Provide environment information

```bash
Operating System:
  Platform: linux
  Arch: x64
  Version: #1 SMP Fri Apr 2 22:23:49 UTC 2021
  Available memory (MB): 15701
  Available CPU cores: 8
Binaries:
  Node: 20.13.1
  npm: 10.5.2
  Yarn: 1.22.22
  pnpm: N/A
Relevant Packages:
  next: 14.2.3 // Latest available version is detected (14.2.3).
  eslint-config-next: 14.2.3
  react: 18.2.0
  react-dom: 18.2.0
  typescript: 5.4.5
Next.js Config:
  output: N/A
```

### Which area(s) are affected? (Select all that apply)

Not sure, Webpack

### Which stage(s) are affected? (Select all that apply)

next dev (local), next build (local), next start (local)

**Comment by icyJoseph** [comment]
Related issue (?) https://github.com/feross/buffer/issues/359

**Comment by fraitag** [comment]
Probably yes. However, I am unsure about the specific version included in the Next.js polyfill. This library is precompiled and is not a Node.js package.

**Comment by icyJoseph** [comment]
I'm guessing 5.x by looking at the compiled/vendored dependency.

On another note, why do you need Buffer in the browser? It is often a sign that code you don't mean to ship to the client, is making its way there, often through browserify

**Comment by fraitag** [comment]
Absolutely, but the external library I use has protection against the lack of Buffer in the environment and executes the logic according to a different path. However, the nextjs mechanisms detect the presence of the word "Buffer" in the external module's code and add a polyfill (though it is not required because the library protects itself against it).

Which does not change the fact that since NextJS gives information that it supports Safari 12+, it also expects it.

**Comment by icyJoseph** [comment]
Buffer doesn't exist in browsers though. 

It is not just Buffer usage that triggers browserification of Buffer, often usage of crypto, or some Node.js modules, does as well. 

Kind of a gray area situation here, can you try out on Safari 14, if the latest buffer (v6) npm package, does work? does v4 work?

**Comment by fraitag** [comment]
Unfortunately, I wasn’t able to build Next.js correctly when I swapped out the Buffer version. I need more time for that. Would anyone else be willing to help?

**Comment by icyJoseph** [comment]
Err, I meant more like:

```html
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>Document</title>
</head>

<body>
    <script src="https://bundle.run/buffer@4.9.2"></script>
    <div>Hi</div>
</body>

</html>
```

And open that in your Safari 14 browser. Try with `6.0.3` and `5.7.1`, as well. Do you see the issue at all?

**Comment by fraitag** [comment]
From bundle.run I do not have any errors, tested on Mojave Safari 12.1
![image](https://github.com/vercel/next.js/assets/17552716/1cfd332e-b914-49cc-a86a-a1e183d069da)

**Comment by icyJoseph** [comment]
Oh that's really, not what I had expected... 🤔 

I searched for the compile PR from 3 years ago, and found that it was ` "buffer": "5.6.0",` that was initially vendored in:

https://github.com/vercel/next.js/pull/32627/files#diff-9c1a3867443c54525b4f24ef171f231a6e8bb065ffc8b7b62c4843d5ff62dd42R81

but that worked on your test...

If you copy the code from, https://github.com/vercel/next.js/blob/canary/packages/next/src/compiled/buffer/index.js#L5, and put it in a file.js and make a script pointing to it, rather than bundle.run, what do you get?

**Comment by philwolstenholme** [comment]
I've just ran into this too while trying to optimise my bundle size and switch to https://es-toolkit.slash.page.

Because of the Buffer polyfill being added our bundle size has actually _increased_!

![image](https://github.com/user-attachments/assets/7e23837b-148f-4e84-a406-f3586d308a64)

**Comment by marcoshernanz** [comment]
Thanks for the report. We could not reproduce this with the original `typeof Buffer` example on Next.js 14.2.3 or the latest canary, and Next’s vendored Buffer 5.6.0 does not contain the BigInt syntax implicated by the linked upstream issue. Next.js 14 is now unsupported, and current Next.js supports Safari 16.4 and newer. If this still occurs in a supported browser on the latest canary, please open a new issue with a stable reproduction.
